### PR TITLE
Feature/add topic tag fields

### PR DESCRIPTION
**Pull request by vmilitaru**
### What

Updates the dataset response model from the dataset-api to include the `CanonicalTopic` and `SubTopics` fields.

### How to review

Read the code and run the tests.

### Who can review

Anyone

### CODE REVIEW

File: dataset/data.go [hunk 1]
```diff
@@ -496,7 +503,12 @@ func (m Metadata) ToString() string {
 	if m.RelatedDatasets != nil {
 		b.WriteString(fmt.Sprintf("Related Links: %s\n", *m.RelatedDatasets))
 	}
-
+	if m.CanonicalTopic != nil{
+		b.WriteString(fmt.Sprintf("Canonical Topic: %s\n", *m.CanonicalTopic))
+	}
+	if m.SubTopics != nil {
```
Comment (rafahop) [inline dataset/data.go:509]: Should we check the length of the array here too? ie. if empty, don't display it? 

Does the `Subtopics` need to be a pointer anyway?

File: dataset/data.go [hunk 2]
```diff
@@ -197,6 +199,11 @@ type Metadata struct {
 	DatasetLinks Links `json:"dataset_links,omitempty"`
 }
 
+type TopicTagObject struct {
```
Comment (rafahop) [inline dataset/data.go:202]: Maybe we could find a better name for this type. Is there any benefit in having `Object` in it? How about simplifying it to `Topic`?

Comment (vmilitaru) [reply dataset/data.go]: `Topic` it is then :white_check_mark:

File: dataset/data.go (hunk 1 above)
Comment (vmilitaru) [reply dataset/data.go:509]: Not really, just following the pattern, but I've changed this now :+1:

**Review by rafahop — APPROVED** [review]

**Pull request merged**